learn/guides/calling-java-code-from-ballerina.md

```diff
@@ -0,0 +1,290 @@
+---
+
+## Writing Ballerina Bindings
+We have introduced a CLI tool called `bindgen` to ease the process of generating Ballerina bindings for given Java APIs. This guide shows you how to use it, if you need more details on how the bindgen tool works, you can refer its reference guide [here](/learn/java-interoperability/the-bindgen-tool/).
+
+The [Package Layout Guide](/learn/organizing-ballerina-code/package-layout/) explains how to package Java libraries (JAR files) with Ballerina programs. When you generate bindings for a Java library using the bindgen tool, this part is already handled. However, you could gain an in-depth understanding of how Java libraries and their transitive dependencies are packaged with Ballerina programs to produce self-contained executable programs.
+
+The [Ballerina FFI Guide](/learn/java-interoperability/ballerina-ffi/) explains how to write these bindings manually. It is also a useful section for those who want to understand the inner workings of calling Java from Ballerina and for those who want to customize the bindings generated by the bindgen tool.
+
+## Using the SnakeYAML Java Library in Ballerina
+SnakeYAML is a YAML parser for Java. In this section, we'll learn how to use this library to parse a YAML document using Ballerina. 
+
+We'll develop a Ballerina program that parses the given YAML file and writes the content to the standard out.
+
+Let's get started.
+
+### Step 1 - Writing the Java Code
+It is recommended to always start by writing the Java code. It gives you an idea of the set of Java classes required to implement your logic. Then, we can use the `bindgen` tool to generate Ballerina bindings for those classes. 
+
+The Java code below uses the SnakeYAML API to parse the given YAML file. Note that this is not the most idiomatic way of writing the Java code for this scenario. 
+
+```java
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.util.Map;
+
+public class SnakeYamlSample {
+
+    public static void main(String... a) {
+	    String filename = a[0];
+        try (InputStream inputStream = new FileInputStream(filename)) {
+            Yaml yaml = new Yaml();
+            Map<String, Object> obj = yaml.load(inputStream);
+            System.out.println(obj);
+        } catch (Exception e) {
+            System.err.println("The file '" + filename + "' cannot be loaded. Reason: " + e.getMessage());;
+        }
+    }
+}
+```
+
+Here, we've used four Java classes. 
+- `org.yaml.snakeyaml.Yaml`
+- `java.io.FileInputStream`
+- `java.io.InputStream`
+- `java.util.Map`
+
+You can see them in the imported class list. We encourage you to generate Ballerina bindings for these four classes as a start.  
+
+Now, we'll create an environment for our Ballerina program. 
+
+### Step 2 - Setting Up the Ballerina Package
+
+This section assumes that you have already read the [Package Layout](/learn/user-guide/ballerina-packages/) guide.
```

Comment:
```suggestion
This section assumes that you have already read the [Package Layout](/learn/user-guide/ballerina-packages/) guide.
```
Instead of adding this as step 2 shall we make it an info box as a prerequisite?